packages/nvidia_nat_a2a/src/nat/plugins/a2a/auth/credential_service.py

```
@@ -0,0 +1,426 @@
+                # Call NAT auth provider (provider is responsible for token refresh/validity)
+                auth_result = await self._auth_provider.authenticate(user_id=user_id)
+
+                # Cache the result
+                self._cached_auth_result = auth_result
+
+                # Warn if provider returned expired credentials (provider bug)
+                if auth_result and auth_result.is_expired():
+                    logger.warning("Auth provider returned already-expired credentials. "
+                                   "This may indicate a bug in the auth provider's token refresh logic.")
+
+            return auth_result
```

Ruff (0.14.8)

Consider moving this statement to an `else` block

(TRY300)